pydocx/export/html.py

```diff
@@ -142,268 +182,470 @@ def style(self):
+
+    def export_footnote(self, footnote):
+        results = super(PyDocXHTMLExporter, self).export_footnote(footnote)
+        tag = HtmlTag('li')
+        return tag.apply(results, allow_empty=False)
+
+    def get_paragraph_tag(self, paragraph):
+        heading_style = paragraph.heading_style
+        if heading_style:
+            tag = self.get_heading_tag(heading_style)
+            if tag:
+                return tag
+        if self.in_table_cell:
+            return
+        if paragraph.has_structured_document_parent():
+            return
+        if isinstance(paragraph.parent, NumberingItem):
+            return
+        return HtmlTag('p')
+
+    def get_heading_tag(self, heading_style):
+        tag = self.heading_level_conversion_map.get(
+            heading_style.name.lower(),
             self.default_heading_level,
         )
-        return self.make_element(
-            tag=heading_value,
-            contents=text,
-        )
+        return HtmlTag(tag)
 
-    def insertion(self, text, author, date):
-        return self.make_element(
-            tag='span',
-            contents=text,
-            attrs={
-                'class': 'pydocx-insert',
-            },
-        )
+    def export_paragraph(self, paragraph):
+        results = super(PyDocXHTMLExporter, self).export_paragraph(paragraph)
 
-    def hyperlink(self, text, href):
-        if text == '':
-            return ''
-        return self.make_element(
-            tag='a',
-            contents=text,
-            attrs={
-                'href': href,
-            },
-        )
+        tag = self.get_paragraph_tag(paragraph)
+        if tag:
+            results = tag.apply(results, allow_empty=False)
 
-    def image_handler(self, image_data, filename, uri_is_external):
-        if uri_is_external:
-            return image_data
-        extension = filename.split('.')[-1].lower()
-        b64_encoded_src = 'data:image/%s;base64,%s' % (
-            extension,
-            base64.b64encode(image_data).decode(),
-        )
-        b64_encoded_src = self.escape(b64_encoded_src)
-        return b64_encoded_src
-
-    def image(self, image_data, filename, x, y, uri_is_external):
-        src = self.image_handler(image_data, filename, uri_is_external)
-        if not src:
-            return ''
-        if all([x, y]):
-            return '<img src="%s" height="%s" width="%s" />' % (
-                src,
-                y,
-                x,
-            )
-        else:
-            return '<img src="%s" />' % src
-
-    def deletion(self, text, author, date):
-        return self.make_element(
-            tag='span',
-            contents=text,
-            attrs={
-                'class': 'pydocx-delete',
-            },
-        )
-
-    def list_element(self, text):
-        return self.make_element(
-            tag='li',
-            contents=text,
-        )
+        for result in results:
+            yield result
```

Comment:
@jlward HTML export_paragraph is SUPER simple now (remember the numbering_level_begin, and the line_break stuff?)